heron/api/src/java/com/twitter/heron/api/topology/UpdatableComponent.java

```diff
@@ -0,0 +1,29 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.twitter.heron.api.topology;
+
+import com.twitter.heron.classification.InterfaceAudience;
+import com.twitter.heron.classification.InterfaceStability;
+
+/**
+ * Bolt or spout instances should implement this method if they wish to be informed of changed to
+ * the topology context during the lifecycle of the instance. The TopologyContext is backed by
+ * elements of the physical plan, which can change during a scaling event. Instances that need to
+ * adapt to such changes should implement this interface to receive a callback when a change occurs.
+ */
+@InterfaceAudience.Public
+@InterfaceStability.Evolving
+public interface UpdatableComponent {
```

Comment:
Instead of calling it UpdatableComponent should we call it IUpdatable to keep interface names consistent.